### PR TITLE
my first refactoring session

### PR DESCRIPTION
**Pull request by cp7781**
be warned though, several days ago an account named skylot appeared out of nowhere on my profile stars;
I contacted support, but they couldn't tell me anything but to change my password;
here, have a look: https://github.com/skylot
someone, who has a decompiler as project; oh my god, that's far beyond my skill level

### CODE REVIEW

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 1]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
-                        setWrappers(commit, wrapper);
+                commits.forEach(commit -> SPIAccessor.DEFAULT.check(commit, false));
+                UndoableWrapper undoableWrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
+                commits.forEach(commit -> {
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(true, this);
                     }
-
-                    if(commit instanceof ProgressProvider) {
+                    if (commit instanceof ProgressProvider) {
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Nice rework of this part

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 2]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
-                    SPIAccessor.DEFAULT.sum(commit);
-                }
+                commits.forEach(commit -> SPIAccessor.DEFAULT.sum(commit));
             }
             if (saveAfterDone) {
                 LifecycleManager.getDefault().saveAll();
-                for (DataObject dob:DataObject.getRegistry().getModified()) {
-                    SaveCookie cookie = dob.getCookie(SaveCookie.class);
+                for (DataObject dataObject : DataObject.getRegistry().getModified()) {
+                    SaveCookie cookie = dataObject.getLookup().lookup(SaveCookie.class);
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: There was a change here:

Before: `dataObject.getCookie(SaveCookie.class)`

After: `dataObject.getLookup().lookup(SaveCookie.class)`

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 3]
```diff
@@ -210,34 +189,32 @@ public void start(ProgressEvent event) {
...
         public void step(ProgressEvent event) {
-            current = current + progressStep;
+            current += progressStep;
             fireProgressListenerStep((int) current);
         }
 
         @Override
         public void stop(ProgressEvent event) {
-            // do not rely on plugins;
+            /* do not rely on plugins; */
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Why the change to a multiline comment?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 4]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
+                        } finally {
                             progressProvider.removeProgressListener(progressListener);
                         }
+                    } else {
+                        commit.commit();
                     }
-                    if (wrapper != null) {
-                        unsetWrappers(commit, wrapper);
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(false, null);
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Should `if (undoableWrapper != null) { undoableWrapper.setActive(false, null); }` be within a finally block?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 5]
```diff
@@ -246,27 +223,20 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
...
                 final Transaction commit = commitIterator.previous();
-                setWrappers(commit, wrapper);
+                undoableWrapper.setActive(true, this);
                 commit.rollback();
-                unsetWrappers(commit, wrapper);
-            }
-            wrapper.close();
-            for (Transaction commit : SPIAccessor.DEFAULT.getCommits(bag)) {
-                SPIAccessor.DEFAULT.sum(commit);
+                undoableWrapper.setActive(false, null);
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Similar to before, should `undoableWrapper.setActive(false, null);` be within a finally block?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 6]
```diff
@@ -246,27 +223,20 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
...
             }
-
-            while (it.hasPrevious()) {
+            undoableWrapper.close();
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Should `undoableWrapper.close();` be within a finally block?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 7]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
+                        commit.commit();
                     }
-                    if (wrapper != null) {
-                        unsetWrappers(commit, wrapper);
+                    if (undoableWrapper != null) {
+                        undoableWrapper.setActive(false, null);
                     }
+                });
+                if (undoableWrapper != null) {
+                    undoableWrapper.close();
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Should `if (undoableWrapper != null) { undoableWrapper.close(); }` be within a finally block?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 8]
```diff
@@ -82,29 +80,23 @@ public static RefactoringSession create(@NonNull String description) {
...
-        });
+        return Utilities.runWithOnSaveTasksDisabled(() -> reallyDoRefactoring(saveAfterDone));
     }
-    
+
     private Problem reallyDoRefactoring(boolean saveAfterDone) {
-        long time = System.currentTimeMillis();
-        
+        final long start = System.currentTimeMillis();
         Iterator it = internalList.iterator();
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: The type of `it` could be `Iterator<RefactoringElementImplementation>`.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 9]
```diff
@@ -210,34 +189,32 @@ public void start(ProgressEvent event) {
...
-        });
+        return Utilities.runWithOnSaveTasksDisabled(() -> reallyUndoRefactoring(saveAfterDone));
     }
-    
+
     private Problem reallyUndoRefactoring(boolean saveAfterDone) {
         try {
-            ListIterator it = internalList.listIterator(internalList.size());
-            fireProgressListenerStart(0, internalList.size()+1);
+            ListIterator internalListIterator = internalList.listIterator(internalList.size());
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: The type of `internalListIterator` could be `ListIterator<RefactoringElementImplementation>`. This should allow the type cast a few lines down to be removed.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 10]
```diff
@@ -246,27 +223,20 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
...
-            while (it.hasPrevious()) {
+            undoableWrapper.close();
+            SPIAccessor.DEFAULT.getCommits(bag).forEach((commit) -> SPIAccessor.DEFAULT.sum(commit));
+            while (internalListIterator.hasPrevious()) {
                 fireProgressListenerStep();
-                RefactoringElementImplementation element = (RefactoringElementImplementation) it.previous();
-                if (element.isEnabled() && !((element.getStatus() == RefactoringElement.GUARDED) || (element.getStatus() == RefactoringElement.READ_ONLY))) {
-                    element.undoChange();
-                }
+                RefactoringElementImplementation element = (RefactoringElementImplementation) internalListIterator.previous();
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: The type cast to `RefactoringElementImplementation` could be removed.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 11]
```diff
@@ -352,61 +341,34 @@ private void fireProgressListenerStop() {
...
-    }
-
-    
     private class ElementsCollection extends AbstractCollection<RefactoringElement> {
+
         @Override
         public Iterator<RefactoringElement> iterator() {
             return new Iterator() {
-                //private final Iterator<RefactoringElementImplementation> inner = internalList.iterator();
+
```
Comment (dtrebbien) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Maybe delete the trailing spaces as well?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 8 above)
Comment (vieiro) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: It's ok as it is.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 4 above)
Comment (vieiro) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: It's ok as it is.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 2 above)
Comment (vieiro) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: getLookup().lookup(SaveCookie.class) is the proper way.

Comment (dtrebbien) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Oh, okay. Corroborating, I see that NetBeans provides an automatic fix which rewrites the getCookie() call to `getLookup().lookup(SaveCookie.class)`.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 12]
```diff
@@ -256,6 +248,19 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
         return null;
     }
 
+    private void changeRefactoringElement(RefactoringElementImplementation element, boolean change) {
+        boolean enabled = element.isEnabled();
+        boolean guarded = element.getStatus() == RefactoringElement.GUARDED;
+        boolean readOnly = element.getStatus() == RefactoringElement.READ_ONLY;
+        if (enabled && !(guarded || readOnly)) {
```
Comment (vieiro) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Hi Carsten, this computes "enabled", "guarded" and "readOnly" even when "element.isEnabled()" is false. The original expression 
`element.isEnabled() && !((element.getStatus() == RefactoringElement.GUARDED) || (element.getStatus() == RefactoringElement.READ_ONLY))`
does not, though.

**Comment by vieiro** [comment]
Hi @cp7781. Thanks for the PR. Are these changes related to an issue? if so then would you please add an issue to JIRA?
Thanks,
Antonio

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 3 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: If using the toggle comment feature with a hotkey, it will use single line comments on each line of the whole selected block. So, the thought was, that a multi-line comment would better reflect the true nature of a comment, and not just some blending out.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 5 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: It was not before. Probably it will be of use, if you explain it to me.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 7 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: There are several nested try and finally block in the source. I haven't touched the overall structure of it.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 9 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: It shall be done. Thank you!

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 6 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Probably. Under the circumstance, that the code from within line 228 to 232 throws an unexpected exception. I'm not sure, if that is even possible. But it is a potential pitfall. I like your approach.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 10 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Well spotted. Not bad!

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 12 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Right, vieiro. I wanted to improve readability, and it resulted in a performance decrease. Haven't even noticed. My bad. Performance is also affected by the added private function call. Before the change, it was done with a redundant inlining of code. Which isn't to bad, but it is not easy to understand.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 11 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: It's a new anonymous inner class implementation, right? The members can be separated by a newline from the enclosing curly braces. But they don't have to?

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 5 above)
Comment (dtrebbien) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Sorry, I was unclear. By "similar to before", I meant "similar to [my previous comment](https://github.com/apache/incubator-netbeans/pull/361#discussion_r161065679) up above".

Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: My apologies. I'm not able to answer that question. I would add things without concrete knowledge about what I'm actually doing.

Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Which is very true.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 11 above)
Comment (dtrebbien) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: I'm fine with the blank lines. It just looked like you were deleting end-of-line whitespace, so the "                " spaces could be deleted as well.

**Comment by cp7781** [comment]
Thank you for speaking for me, Antonio Vieiro.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 3 above)
Comment (svatoun) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Line comment was specifically invented for realy short notes and it's faster to type than /* - */ pair. I don't see a point to change it for one-liners. Maybe for long multilines.
It would be more useful to extend the toggle comment feature to:
- make a block comment (maybe a different hotkey or an additional modifier)
- turn block of line comments into block comment or vice versa

Whether to use multilines or block comments depend on personal or group development style - it's not a defect to fix.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 13]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
                         Exceptions.printStackTrace(ex);
                     }
                 }
             }
-            for (RefactoringElementImplementation fileChange:SPIAccessor.DEFAULT.getFileChanges(bag)) {
-                if (fileChange.isEnabled()) {
-                    fileChange.performChange();
-                }
-            }
+            SPIAccessor.DEFAULT.getFileChanges(bag).stream().filter(fileChange -> fileChange.isEnabled()).forEachOrdered(fileChange -> fileChange.performChange());
```
Comment (svatoun) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Well, I sort of doubt that this stream syntax improves readability over (more verbose) `for` or `forEach`. But please make a breakline between the individual processing steps, like this:
```
SPIAccessor.DEFAULT.getFileChanges(bag).stream().
    filter(fileChange -> fileChange.isEnabled()).
    forEachOrdered(fileChange -> fileChange.performChange());
```

Which can be usually scanned faster. Debugging of streamized code is a real pain, BTW.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 14]
```diff
@@ -116,79 +108,66 @@ private Problem reallyDoRefactoring(boolean saveAfterDone) {
...
-                realcommit=false;
+                realcommit = false;
             }
         }
         Logger timer = Logger.getLogger("TIMER.RefactoringSession");
         if (timer.isLoggable(Level.FINE)) {
-            time = System.currentTimeMillis() - time;
-            timer.log(Level.FINE, "refactoringSession.doRefactoring", new Object[] { description, RefactoringSession.this, time } );
+            final long timeTaken = System.currentTimeMillis() - start;
+            timer.log(Level.FINE, "refactoringSession.doRefactoring", new Object[]{description, RefactoringSession.this, timeTaken});
```
Comment (svatoun) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Considering most of the PR fixes whitespaces, there should be whitespaces before/after { to separate from type and array content, and before }.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 15]
```diff
@@ -246,27 +223,20 @@ private Problem reallyUndoRefactoring(boolean saveAfterDone) {
                     f.undoChange();
                 }
             }
-            for (Transaction commit : SPIAccessor.DEFAULT.getCommits(bag)) {
-                SPIAccessor.DEFAULT.check(commit, true);
-            }
-            UndoableWrapper wrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
+            SPIAccessor.DEFAULT.getCommits(bag).forEach(commit -> SPIAccessor.DEFAULT.check(commit, true));
+            UndoableWrapper undoableWrapper = MimeLookup.getLookup("").lookup(UndoableWrapper.class);
```
Comment (svatoun) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: // NOI18N marker missing.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java [hunk 16]
```diff
@@ -352,61 +341,34 @@ private void fireProgressListenerStop() {
...
                 public void remove() {
                     throw new UnsupportedOperationException();
                 }
-                
+
                 @Override
                 public RefactoringElement next() {
                     if (index < internalList.size()) {
-                        return new RefactoringElement(internalList.get(index++));
+                        RefactoringElementImplementation implementation = internalList.get(index);
```
Comment (svatoun) [inline refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Coding style in effect does not prohibit pre/postfix increments.

**Comment by geertjanw** [comment]
A lot of people are spending a lot of time reviewing this pull request. It is unclear why this pull request exists. It seems this is a cosmetic refactoring of a random file in the Apache NetBeans Git repo. We do need to set up guidelines for pull requests and surely one of those will be that we need to have an issue number related to a pull request, i.e., an actual problem that we'd like to have fixed, before fixing it.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 13 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: You are right. Would prefer, if the dot is placed at the new line, in front of the method call.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 14 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: I don't know, what a PR fix is. But to separate things, to get a better overview, seems to be a good idea.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 15 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: There is no string, which needs to be translated. Why adding a comment?

**Comment by cp7781** [comment]
The project needs far more developers refactoring their work constantly. Especially at the core product. If the overall structure crumbles and falls, it will be just another waste of time in the history of mankind.

**Comment by dtrebbien** [comment]
I think that fixing all of the warnings actually helps. When bug fixing, it is distracting to see all of the yellow warning indicator lines in the sidebar. It makes it more difficult to see what you've changed, as the light green, gray, and pink added/changed/deleted indicator lines are hidden by the yellow lines. It also makes it more difficult to see the locations of breakpoints that you have set. For example, here is the bar for a file in which I have set two breakpoints and made changes to two parts of the file:
screen shot 2018-01-12 at 8 51 27 am

One other benefit to general cleanup (in particular, supplying type information on generics) is that it cuts down on the voluminous output of build warnings.

I have experimented with submitting the "cleanups" in the same patch as a bug fix (e.g. PR #287), but I don't like this approach. One must be in the mood for examining a lot of small changes that result from cleanup. This work is distracting from reviewing the meat of the patch.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 13 above)
Comment (dtrebbien) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: Right. I think that placing the dot first and indenting twice is more standard:

```java
SPIAccessor.DEFAULT.getFileChanges(bag).stream()
        .filter(fileChange -> fileChange.isEnabled())
        .forEachOrdered(fileChange -> fileChange.performChange());
```

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 15 above)
Comment (dtrebbien) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: The `// NOI18N` indicates that the string does not need to be translated.

I believe that some devs have a plugin which places warnings on all strings that are not marked "NOI18N" as a way of verifying that all strings either do not need translation or are localizable.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 16 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: The IDE complained about it. There is an initialization, a method call, and an increment in one line. Three actions, three lines of code.

Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: I may have seen white rabbits. Changed it back, and no complaints at all. Sometimes, mysterious things happen, and I just try to get used to it, alright? You are right. Thank you for the support.

File: refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java (hunk 3 above)
Comment (cp7781) [reply refactoring.api/src/org/netbeans/modules/refactoring/api/RefactoringSession.java]: A different hotkey is not a good idea for me. I have reconfigured some actions to use the F1 to F12 keys. And since that is just not enough, I have started to use the Ctrl key in combination with some characters. Just so, that I can easily reach them both with only my left hand, and without broken fingers. At this side of the keyboard there aren't much unused keys left for reconfiguration.  And it would feel unnatural to use two different keystrokes for similar actions. The /**/ commentation mode is already used for documentation. Why not keep it like that, and use // mostly for partial code changes? Since "do not rely on plugins;" can not be evaluated by the compiler, I would recomment using /**/ commentation mode.

**Comment by vieiro** [comment]
Hi,
Thanks for this PR. These changes seem to improve readability and maintainability, and not a real user problem, so I'm adding the "Cosmetics" label.
I'm also adding the "work in progress" label. Please request a review whenever you're done.
Thanks,
Antonio

**Comment by cp7781** [comment]
Cosmetics sound pretty good. Don't they? It's not just fonts and stuff. Good morning, spain.

**Comment by cp7781** [comment]
Can anyone tell me, were to get rid of that unit tests? I mean, now it looks like there is a reference somewhere, which needs to be fixed. And actually I don't want to write unit tests for unit test routines, just because it means business.

**Comment by cp7781** [comment]
For now, I wouldn't mind those tests, Daniel. There is some package missing, or something. And it turns red, if the plugin is opened. Just want to fix that.

* This computation of the `count` variable indeed looks really awkward. What exactly is getting computed?

  ```java
  {
      int count = COMMIT_STEPS + commits.size() * COMMIT_STEPS + 1;
      fireProgressListenerStart(0, count);
  }
  ```
  The `fireProgressListenerStep()` method is getting called at line 117 and 163. Once in a loop over items in the `internalList` member variable, and once at the end of a try block. There seems to be no connection to the call of `commits.size()` in the code listing above. So, this part might be corrupted?

* Using lambda expressions is an elegant way to streamline code. Before change, there were two private functions in place, which were both called just once. It not only clutters the navigator window, it also distracts attention away from the requirements.

* Thank you for mentioning `Stream.of(...)`.

* Doctor, can you please help me? The people say, I'm an idiot, and they are constantly laughing at me.

Order of maw means breakfast. I had toast, honey and milk. And a nice girl was talking to me meanwhile. Life is just to good for waking up early.

**Comment by sdedic** [comment]
Keep in mind that usage of Lambda expression also totally breaks fix-and-continue during debugging. It's a nice syntax suggar but until Hotspot folks fix the VM, class with lambdas does not allow to incrementally fix the code.

As for unit tests: they run on the master. So the only sane approach is to fix the code so that tests run again. But this PR is WIP, so OK. But before the PR is reviewed, I recommend to discard that commit from history.

The PR starts being hard to read, mixing (intentionally just) formatting changes with refactorings which actually have potential to break something. I recommend to separate those two actions into either separate commits, or separate PRs - it's easier to verify that refactoring did not change semantic if 90% of file is not whitespace-changed.

As for commit messages (personal opinion): I will not approve PR with messages like "for the queen", "fixing errors" (which actually deletes test code), or "order of maw". I believe that added confusion which outweights the fun of reading them.

**Comment by cp7781** [comment]
Debugging, like making use of assert statements, is a relic of the past. Modern software in general has become far way to complex for being used with such features. But take it as my humble opinion. I've almost no idea, how a real project would look like, and what possibly can be done, if used cleverly by a true master.

**Comment by cp7781** [comment]
So, just to make sure. The tags `code-cleanup` and `work-in-progress` seem to mark somewhat suspicious pull requests. You likely want to concentrate on other things; probably more useful stuff, than review and education. I will close this. Good luck!
